pkg/redis/client.go

```diff
@@ -0,0 +1,188 @@
+package redis
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+// ErrNotFound indicates that the record is not found in redis.
+var (
+	ErrNotFound = errors.New("redis: not found")
+)
+
+// Client is the public interface so that implementations can be mocked or swapped.
+type Client interface {
+	GetSandboxBySessionID(ctx context.Context, sessionID string) (*Sandbox, error)
+	SetSessionLockIfAbsent(ctx context.Context, sessionID string, ttl time.Duration) (bool, error)
+	BindSessionWithSandbox(ctx context.Context, sessionID string, sb *Sandbox, ttl time.Duration) error
+	DeleteSessionBySandboxIDTx(ctx context.Context, sandboxID string) error
+}
+
+// client is a go-redis based implementation; it is internal and only exposed via Client.
+type client struct {
+	rdb           *redis.Client
+	sessionPrefix string
+	sandboxPrefix string
+	lockPrefix    string
+}
+
+// Option configures client behavior (e.g. key prefixes).
+type Option func(*client)
+
+// WithKeyPrefixes sets custom key prefixes for session / sandbox / lock.
+func WithKeyPrefixes(session, sandbox, lock string) Option {
+	return func(c *client) {
+		c.sessionPrefix = session
+		c.sandboxPrefix = sandbox
+		c.lockPrefix = lock
+	}
+}
+
+// NewClient creates a new Client using the given go-redis client.
+func NewClient(rdb *redis.Client, opts ...Option) Client {
```

Comment:
I am thinking you should init the redis client here instead of passing in. This is the responsibility of redis package